plugins/outputs/cloudwatchlogs/cloudwatchlogs.go

```diff
@@ -79,22 +82,25 @@ func (c *CloudWatchLogs) Write(metrics []telegraf.Metric) error {
 	return nil
 }
 
-func (c *CloudWatchLogs) CreateDest(group, stream string) logs.LogDest {
+func (c *CloudWatchLogs) CreateDest(group, stream string, retention int) logs.LogDest {
 	if group == "" {
 		group = c.LogGroupName
 	}
 	if stream == "" {
 		stream = c.LogStreamName
 	}
+	if retention <= 0 {
+		retention = -1
+	}
 
 	t := Target{
 		Group:  group,
 		Stream: stream,
 	}
-	return c.getDest(t)
+	return c.getDest(t, retention)
```

Comment:
Agreed. I don't see any reason you wouldn't just put the retention int in the Target struct. If the log group and log stream were separate parameters for the `getDest()` function, I could see why you wouldn't, but since they're being wrapped in the struct, the retention period should as well.